node/http/src/api.rs

```diff
@@ -62,11 +62,15 @@ impl HttpApi {
             r.contract_account_id,
             r.method_name,
         );
-        let best_block = self.client.shard_chain.chain.best_block();
+        let state_update = self.client.shard_chain.get_state_update();
```

Comment:
we are continuing to add more logic like this to the network interface layer .. this kind of stuff should be wrapped by the client cc: @nearmax

Reply:
Actually `state_update` is something that client will eventually expose, but you are right a lot of stuff will be wrapped into the client.

Reply:
https://github.com/nearprotocol/nearcore/issues/536

Reply:
It's not really a client if it's not needed for operation of the protocol IMO. A view later that is shared between different accessors like RPC, we sockets etc is fine.

Reply:
yea.. we've hinted on that here https://github.com/nearprotocol/nearcore/issues/479

maybe #536 is a dupe

Reply:
A reasoning is the client will need to be triple reviewed / bug bounty and etc vs extra code that is not involved in operation of protocol can be more relaxed.

Reply:
> A reasoning is the client will need to be triple reviewed / bug bounty and etc vs extra code that is not involved in operation of protocol can be more relaxed.

yea, feel like that was the motivation behind #479

Reply:
I like the idea of separating view layer from the domain logic, if the only thing view layer does is providing convenience accessors to stuff inside the client, and it does not mutate client's state.